pkg/resources/masking_policy_acceptance_test.go

```diff
@@ -2,79 +2,300 @@ package resources_test
+					HasNameString(id.Name()).
+					HasDatabaseString(id.DatabaseName()).
+					HasSchemaString(id.SchemaName()).
+					HasReturnDataTypeString(string(sdk.DataTypeVARCHAR)).
+					HasFullyQualifiedNameString(id.FullyQualifiedName()).
+					HasBodyString(body).
+					HasExemptOtherPoliciesString(r.BooleanDefault).
+					HasArguments(argument),
+				),
+			},
+			// set all fields
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_MaskingPolicy/complete"),
+				ConfigVariables: tfconfig.ConfigVariablesFromModel(t, policyModel.WithBody(body).WithComment("Terraform acceptance test").WithExemptOtherPolicies(r.BooleanTrue)),
+				Check: assert.AssertThat(t, resourceassert.MaskingPolicyResource(t, resourceName).
+					HasNameString(id.Name()).
+					HasDatabaseString(id.DatabaseName()).
+					HasSchemaString(id.SchemaName()).
+					HasExemptOtherPoliciesString(r.BooleanTrue).
+					HasReturnDataTypeString(string(sdk.DataTypeVARCHAR)).
+					HasFullyQualifiedNameString(id.FullyQualifiedName()).
+					HasCommentString("Terraform acceptance test").
+					HasBodyString(body).
+					HasArguments(argument),
+					resourceshowoutputassert.MaskingPolicyShowOutput(t, resourceName).
+						HasCreatedOnNotEmpty().
+						HasDatabaseName(id.DatabaseName()).
+						HasKind(string(sdk.PolicyKindMaskingPolicy)).
+						HasName(id.Name()).
+						HasExemptOtherPolicies(true).
+						HasOwner(snowflakeroles.Accountadmin.Name()).
+						HasOwnerRoleType("ROLE").
+						HasSchemaName(id.SchemaName()),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.body", body)),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.name", id.Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.return_type", string(sdk.DataTypeVARCHAR))),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.signature.#", "2")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.signature.0.name", "A")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.signature.0.type", string(sdk.DataTypeVARCHAR))),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.signature.1.name", "B")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.signature.1.type", string(sdk.DataTypeVARCHAR))),
 				),
 			},
-			// rename + change comment
+			// change fields
 			{
-				Config: maskingPolicyConfig(newId.Name(), comment2, acc.TestDatabaseName, acc.TestSchemaName),
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_MaskingPolicy/complete"),
+				ConfigVariables: tfconfig.ConfigVariablesFromModel(t, policyModel.WithBody(bodyWithBooleanReturnType).WithReturnDataType(string(sdk.DataTypeBoolean)).WithArgument(argumentWithChangedFirstArgumentType).WithComment("Terraform acceptance test - changed comment").WithExemptOtherPolicies(r.BooleanFalse)),
```

Comment:
The way that policyModel is reused here is interesting/unsafe - that was not the initial intention to reuse it this way. It bases on the order of execution and is a bit misleading (e.g. in step 4 and step 5 we want to validate only external change but we use "different" config variables - different only in the written way, underneath it's the same pointer, with fields already set in the previous step).

Maybe the design of model is a bit flawed and we should not use the pointer but the model directly, so that we always get the copied struct. Either way, I am raising it as a topic to discuss internally.